egograph/dataset_catalog/validation.py

```
@@ -0,0 +1,79 @@
+
+    pandas の内部 dtype に依存せず、バイト列から schema を取得して検証する。
+    required_columns / column_types の両方を検証し、アップロード前の検証に
+    使うことを想定している。読み直しは行わない。
+
+    Args:
+        definition: 契約元の DatasetDefinition
+        data: アップロード前の Parquet バイト列
+
+    Raises:
+        ValueError: 契約外のカラム・型が見つかった場合
+            （``invalid_schema: ...``）
```

Ruff (0.16.0)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)